### PR TITLE
Bug: Add correct service names for timestream boto3 clients

### PR DESCRIPTION
**Pull request by malachi-constant**
###  Bugfix
- Bug: Add correct service names for timestream boto3 clients

### Detail
- Timestream boto3 client service names are [`timestream-query`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/timestream-query.html#client) and [`timestream-write`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/timestream-write.html#client) and our config method needs to expect those service names instead of `timestream`

### Relates
- #1710 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: f0376f942e3d3366ba3fa8ee609b563a37c7069c
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=feb4763e-3ccd-4420-b67b-2c8b506d823f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/_utils.py [hunk 1]
```diff
@@ -99,8 +99,11 @@ def _get_endpoint_url(service_name: str) -> Optional[str]:
         endpoint_url = _config.config.dynamodb_endpoint_url
     elif service_name == "secretsmanager" and _config.config.secretsmanager_endpoint_url is not None:
         endpoint_url = _config.config.secretsmanager_endpoint_url
-    elif service_name == "timestream" and _config.config.timestream_endpoint_url is not None:
+    elif service_name == "timestream-write" and _config.config.timestream_endpoint_url is not None:
```
Comment (jaidisido) [inline awswrangler/_utils.py:102]: The same change is required here: https://github.com/aws/aws-sdk-pandas/blob/timestream-write/awswrangler/_utils.py#L83

Thanks

Comment (malachi-constant) [reply awswrangler/_utils.py:102]: The link you shared seems to be this same function here where the change is.

Comment (jaidisido) [reply awswrangler/_utils.py:102]: Ah yes my bad, [here](https://github.com/aws/aws-sdk-pandas/blob/timestream-write/awswrangler/_config.py#L51) rather. We might need to set two different endpoint urls, one for write and the other for query

Comment (malachi-constant) [reply awswrangler/_utils.py:102]: Gotcha, yea agreed we should allow for setting both. Will update.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b5c31f429d449a45eff9f793e91e44eb2ce9350d
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7487a22a-32c3-4702-ada6-57cfeb56b45d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
Cannot replicate unit testing errors locally, will rerun job.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: b5c31f429d449a45eff9f793e91e44eb2ce9350d
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=63c06d42-3117-4098-8627-80fcbf950624%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b5c31f429d449a45eff9f793e91e44eb2ce9350d
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=eaa231ed-bb1b-46aa-9b16-e933ebaa5a45%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b5c31f429d449a45eff9f793e91e44eb2ce9350d
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bb58a6c2-3d20-4bb4-8fac-c315e545c922%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — APPROVED** [review]

**Review by kukushking — APPROVED** [review]

**Review by LeonLuttenberger — APPROVED** [review]

**Pull request merged**